Source/WebCore/dom/make_names.pl

```diff
@@ -759,145 +833,552 @@ END
+    print F "#include <wtf/text/AtomString.h>\n";
+    print F "\n";
+    print F "namespace WebCore {\n";
+    print F "\n";
+    print F "class QualifiedName;\n";
+    print F "\n";
+    print F "enum class TagName : uint16_t {\n";
+    print F "    Unknown,\n";
+
+    my %handledTags = ();
+    my $previousCppNamespace = "";
+    my $seenMultiNamespaceTags = 0;
+    for my $elementKey (sort byElementNameOrder keys %allElements) {
+        my $identifier = $allElements{$elementKey}{identifier};
+        next if $handledTags{$identifier};
+
+        my $localName = $allElements{$elementKey}{localName};
+        my $namespace = $allElements{$elementKey}{namespace};
+        my $tagEnumValue = $allElements{$elementKey}{tagEnumValue};
+
+        my $sectionComment =
+            elementCount($localName) > 1 && !$seenMultiNamespaceTags ? "\n    // Tags in multiple namespaces\n" :
+            $allElements{$elementKey}{cppNamespace} ne $previousCppNamespace ? "\n    // $allElements{$elementKey}{cppNamespace}\n":
+            "";
+        my $inlineComment = elementCount($localName) > 1 ? " // " . join(", ", sort keys(%{$allNamespacesPerElementLocalName{$localName}})) : "";
+
+        print F $sectionComment;
+        print F "    $tagEnumValue,$inlineComment\n";
+
+        $seenMultiNamespaceTags = 1 if elementCount($localName) > 1;
+        $previousCppNamespace = $allElements{$elementKey}{cppNamespace};
+        $handledTags{$identifier} = 1;
+    }
+
+    print F "\n";
+    print F "    // Foreign namespace tag names requiring adjustment\n";
+
+    my %namespacesWithTagsRequiringAdjustment = ();
+    for my $elementKey (sort byElementNameOrder keys %allElements) {
+        next if $allElements{$elementKey}{unadjustedTagEnumValue} eq "";
+        print F "    $allElements{$elementKey}{unadjustedTagEnumValue},\n";
+        $namespacesWithTagsRequiringAdjustment{$allElements{$elementKey}{namespace}} = 1;
+    }
+
+    print F "};\n";
+    print F "\n";
+
+    my $lastTagEnumValue;
+    for my $elementKey (sort byElementNameOrder keys %allElements) {
+        my $unadjustedTagEnumValue = $allElements{$elementKey}{unadjustedTagEnumValue};
+        $lastTagEnumValue = $unadjustedTagEnumValue if $unadjustedTagEnumValue ne "";
+    }
+
+    print F "inline constexpr auto lastTagNameEnumValue = TagName::$lastTagEnumValue;\n";
+    print F "inline LazyNeverDestroyed<EnumeratedArray<TagName, AtomString, lastTagNameEnumValue>> tagNameStrings;\n";
+    print F "\n";
+    print F "WEBCORE_EXPORT void initializeTagNameStrings();\n";
+    print F "TagName findTagName(Span<const UChar>);\n";
+    print F "#if ASSERT_ENABLED\n";
+    print F "TagName findTagName(const String&);\n";
+    print F "#endif\n";
+    for my $namespace (sort keys %namespacesWithTagsRequiringAdjustment) {
+        print F "TagName adjust${namespace}TagName(TagName);\n";
+    }
+    print F "\n";
+    print F "inline const AtomString& getTagNameAsString(TagName tagName)\n";
```

Comment:
This should be tagNameAsString or stringForTagName.
"get" prefix should be only used when there is an out argument:
https://webkit.org/code-style-guidelines/#names-setter-getter